appserver/payara-appserver-modules/microprofile/metrics/src/main/java/fish/payara/microprofile/metrics/cdi/MetricsHelper.java

```diff
@@ -168,47 +150,99 @@ private String getParameterName(AnnotatedParameter<?> parameter) {
-            return metadataOf(member, Histogram.class);
+            builder.withType(MetricType.HISTOGRAM);
         } else if (typeName.startsWith(Timer.class.getName())) {
-            return metadataOf(member, Timer.class);
+            builder.withType(MetricType.TIMER);
         }
-        return null;
+        return builder;
     }
-
-    private Metadata metadataOf(AnnotatedMember<?> member, Class<?> type) {
-        return metadataOf(member, type, metricNameOf(member));
+    
+    public MetricID metricIDof(InjectionPoint ip) {
+        Annotated annotated = ip.getAnnotated();
+        if (annotated instanceof AnnotatedMember) {
+            return metricIDof((AnnotatedMember<?>) annotated, ip.getMember().getDeclaringClass());
+        } else if (annotated instanceof AnnotatedParameter) {
+            return metricIDof((AnnotatedParameter<?>) annotated, ip.getMember().getDeclaringClass());
+        } else {
+            throw new IllegalArgumentException("Unable to retrieve metric name for injection point [" + ip + "], only members and parameters are supported");
+        }
     }
-
-    private Metadata metadataOf(Annotated annotated, Class<?> type, String name) {
-        Metadata metadata = new Metadata(name, MetricType.from(type));
-        if (annotated.isAnnotationPresent(Metric.class)) {
-            Metric metric = annotated.getAnnotation(Metric.class);
-            metadata.setDescription(metric.description() == null || metric.description().trim().isEmpty() ? null
-                    : metric.description());
-            metadata.setDisplayName(metric.displayName() == null || metric.displayName().trim().isEmpty() ? null
-                    : metric.displayName());
-            metadata.setUnit(metric.unit() == null || metric.unit().trim().isEmpty() ? null
-                    : metric.unit());
-            for (String tag : metric.tags()) {
-                metadata.addTag(tag);
+    
+    public MetricID metricIDof(Annotated member, Class baseClass) {
+        if (member.isAnnotationPresent(Metric.class)) {
+            Metric metric = member.getAnnotation(Metric.class);
+            if (metric.absolute()) {
+                return new MetricID(metric.name(), tagsFromString(metric.tags()));
+            } else {
+                return new MetricID(baseClass + "." + metric.name(), tagsFromString(metric.tags()));
+            }
+        } 
+        return null;
+    }
+    
+   /*public MetricID metricIDof(AnnotatedMember<?> member) {
+        if (member.isAnnotationPresent(Metric.class)) {
+            Metric metric = member.getAnnotation(Metric.class);
+            return new MetricID(metric.name(), tagsFromString(metric.tags()));
+        }        
+        return null;
+    }*/
+    
+    public static Tag[] tagsFromString(String[] stringtags) {
+        Tag[] tags = new Tag[stringtags.length];
+        for (int i = 0; i < stringtags.length; i++) {
+            int splitIndex = stringtags[i].indexOf('=');
```

Comment:
Could use String.split() here? Would save from needing to collect substrings later on and could just check length of produced array being 1 rather than the splitIndex being -1